middleware/entirehouseCache.js

```diff
@@ -0,0 +1,13 @@
+const client = require('../constants/client');
+
+module.exports.headerCache = (req, res, next) => {
+    const key = `entirehouse${req.params.propertyId}`;
```

Comment:
Addressing all comments here:

1. Outside of the syntax, I had not looked deep into other than knowing node uses CommonJS and the browser supports ES6 modules. Read a little more into it, I saw in multiple sources that ES6 modules supports asynchronous and synchronous importing of modules while CommonJS is only synchronous (loads modules one by one). So my understanding based on that knowledge is that loading modules asynchronously ensures that loading is non-blocking in the browser.

Question: I can wrap my head around the idea that we do not want our code in the browser to be blocked/prevented from loading when there are several files to load and those files can work independently of each other, however I am wondering if there is any benefit from async module loading in a node environment. Doesn't the code on the server NEED to be entirely loaded in order to work?

2. The reasoning for implementing the cache was mostly out of curiosity. I read a bit into how browser caching and caching data on the server works and wanted to see it work for my services. Would you prefer that I remove it? If so, I can just comment out the middleware on my Proxy EC2 instance. To address your comment about the error handling, assuming I can keep the cache, it makes sense to me to handle the error by not handling the error (I mean I can remove the "if (err)" conditional entirely) because if for whatever reason there is an error with the cache, the application will just get the data directly from the database. Or I can console.log() the error and save that log so I can read what is going on, but not effect the flow of the application. What are your thoughts?


I apologize for the lengthy response, but I do appreciate your feedback. Thank you, Leslie.